### DISCUSSION TITLE
Relation when updating: Use the ID directly instead of using `connect`

### DISCUSSION THREAD

**Discussion by bennettdams** [prisma/prisma#14804] [Q&A] [answered]
I wanted to use a referenced ID directly instead of using the `connect` statement.

Here's a basic example to update the author of a post. Each post has an `authorId`, which is a relation to a user model.

```ts
await prisma.post.update({
  where: { postId },
  data: {
    postTitle,
    authorId,
  },
})
```

This would usually look like this (using `author` instead of `authorId`):

```ts
await prisma.post.update({
  where: { postId },
  data: {
    postTitle,
    author: { connect: { authorId } },
  },
})
```

In this example not using `connect` makes it easier to update, as my inputs (`postTitle` and `author`/`authorId`) are optional/undefined, so I don't have to check them for existence first.

This seems to work, but are there downsides? I have not read about this approach in the docs, hence I'm worried there are some lurking issues with it.

**Comment by austincrim** [comment, marked as answer]
Hey @bennettdams 👋,

Specifying a foreign key directly is just fine and will work without issue 👍. The only difference is that `connect` will throw an error if it can't find the specified relation.

**Reply by bennettdams** [reply]
Ahh, that's an important distinction! I would rather use `connect` then. Thanks!

**Reply by bennettdams** [reply]
I just realized that you can have the best of both worlds by using a ternary:

```ts
  data: {
    postTitle,
    author: !authorId ? undefined : { connect: { authorId } },
  },
```

With this ternary you a) don't have to check `authorId` manually, because it will not execute the update if `authorId` is nullish, as Prisma will ignore `undefined` for the update operation and b) still uses `connect` instead, so an error is thrown if it can't find the resource via `authorId`.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
